macros/src/interface/generate.rs

```diff
@@ -296,28 +296,37 @@ fn generate_service(item_tokens: &mut TokenStream, client_impl_tokens: &mut Toke
 	if service.request_updates().is_empty() && service.response_updates().is_empty() {
 		client_impl_tokens.extend(quote! {
 			#service_doc
-			pub async fn #service_name(&self, #request_param) -> Result<#response_type, #fizyr_rpc::error::ServiceCallError>
+			pub async fn #service_name(&self, #request_param) -> Result<#response_type, #fizyr_rpc::Error>
 			where
 				F: #fizyr_rpc::util::format::EncodeBody<#request_type>,
 				F: #fizyr_rpc::util::format::DecodeBody<#response_type>,
 			{
-				let request_body = #request_body.map_err(#fizyr_rpc::error::SendRequestError::EncodeBody)?;
+				let request_body = #request_body.map_err(#fizyr_rpc::Error::encode_failed)?;
 				let mut request = self.peer.send_request(#service_id, request_body).await?;
+
 				let response = request.recv_response().await?;
-				let decoded = F::decode_body(response.body).map_err(#fizyr_rpc::error::RecvMessageError::DecodeBody)?;
-				Ok(decoded)
+				if response.header.service_id == -1 {
```

Comment:
Should we make this a definition somewhere, instead of a "magic" value?

Reply:
I was wondering if you would suggest this :p I think there even is a constant for it already.

Reply:
Updated to use `fizyr_rpc::service_id::ERROR`.

Reply:
Is this a test or a review ? :P

Reply:
Both? :p 

Or just a test to see how lazy I can be..